### PR TITLE
(pouchdb/pouchdb#3410) - use pouchdb-upsert

### PR DESCRIPTION
**Pull request by nolanlawson**

### CODE REVIEW

**Comment by nolanlawson** [comment]
Well, this actually increases the min+gz size a tiny bit when combined with https://github.com/pouchdb/mapreduce/pull/249 (from 45314 to 45443), probably due to the `putIfNotExists` function, which we're not using. But whatever, the code's cleaner this way. :stuck_out_tongue:

File: test/test.js [hunk 1]
```diff
@@ -54,7 +54,7 @@ describe('upsert', function () {
       }
     }, 'foo', function () {
       return false;
-    }).should.become('lalala');
+    }).should.be.fulfilled;
```
Comment (nolanlawson) [inline test/test.js]: upsert actually resolves with an object like `{updated: true}` now

**Comment by daleharvey** [comment]
+1 looks good

**Comment by nolanlawson** [comment]
1f67e2fc082cc138fccec3bbb9797d61a33c00c5

**Pull request merged**